From 447e2b3cec85c0d3b95d4db4cdaf543f0b9f7b44 Mon Sep 17 00:00:00 2001
From: Gu, Wangyi <wangyi.gu@intel.com>
Date: Wed, 22 May 2013 13:23:07 +0800
Subject: fix the messy issue when play the exported video in Merrifield platform(2)

BZ:96353

root cause: the first SPS/PPS are not stored together with first IDR frame from the
encoder, and parameters of SPS/PPS in each sequence are not always same. This patch
adds one more SPS/PPS together with the first IDR frame.
This patch should work with http://android.intel.com:8080/#/c/108935/

Category:aosp improvement
Domain: Video.Media-video editor
Origin: internal
Upstream-Candidate: No

Change-Id: I393250f5d226b8b6f130957f392e68bb63475a0d
Orig-Change-Id: I9ab4f5c26b51584f337ffe051d53ff6adecf5616
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 libvideoeditor/vss/mcs/src/M4MCS_API.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libvideoeditor/vss/mcs/src/M4MCS_API.c b/libvideoeditor/vss/mcs/src/M4MCS_API.c
index ab8f887..42717f9 100755
--- a/libvideoeditor/vss/mcs/src/M4MCS_API.c
+++ b/libvideoeditor/vss/mcs/src/M4MCS_API.c
@@ -867,14 +867,14 @@ M4OSA_ERR H264MCS_ProcessEncodedNALU(   M4OSA_Void *ainstance,
         if( nal_unit_type == 8 )
         {
             M4OSA_TRACE1_0("Error : PPS");
-            return 0;
+            continue;
         }
 
         if( nal_unit_type == 7 )
         {
             /*SPS Packet */
             M4OSA_TRACE1_0("Error : SPS");
-            return 0;
+            continue;
         }
 
         if( (nal_unit_type == 5) )
-- 
1.7.4.1

